dashboard/app/controllers/code_reviews_controller.rb

```diff
@@ -0,0 +1,62 @@
+class CodeReviewsController < ApplicationController
+  before_action :authenticate_user!
+
+  # Make sure we don't forget to authorize each action
+  check_authorization except: [:index]
+
+  # GET /code_reviews
+  # Returns the list of code reviews and associated comments for the signed-in
+  # user and the given script and level.
+  def index
+    params.require([:scriptId, :levelId])
+
+    code_reviews = CodeReview.get_all(
+      user_id: current_user.id,
+      script_id: params[:scriptId],
+      level_id: params[:levelId]
+    )
+
+    render json: code_reviews.map(&:summarize_with_comments)
+  end
+
+  # POST /code_reviews
+  def create
+    params.require([:scriptId, :levelId, :channelId, :version])
+
+    storage_id, project_id = storage_decrypt_channel_id(params[:channelId])
+    project_owner_id = user_id_for_storage_id(storage_id)
+    # TODO: Should we check that this is a valid version for this project?
+    # TODO: Can we determine and store an accurate expiration date? Can the expiration date change?
+
+    # TODO: Consider storing the channel_id instead of the project_id in CodeReview
+    code_review = CodeReview.new(
+      user_id: current_user.id,
+      script_id: params[:scriptId],
+      level_id: params[:levelId],
+      project_id: project_id,
+      project_version: params[:version]
+    )
+    authorize! :create, code_review, project_owner_id
+    code_review.save!
+
+    render json: code_review.summarize
+  end
+
+  # PATCH /code_reivews/:id
+  # Currently, closing the code review is the only allowed update.
```

Comment:
Do we need to send an error if other params are sent? Or will it be clear from summarize what's been updated?

Reply:
Hmmm, good question.  It won't be clear if a param was just ignored.  I'll take a quick look, but I may address this later.  Is that ok?

Reply:
Yup!